src/simulators/unitary/unitary_state.hpp

```diff
@@ -382,6 +391,48 @@ void State<data_t>::apply_matrix(const reg_t &qubits, const cvector_t &vmat) {
   }
 }
 
+template <class data_t>
+void State<data_t>::apply_multiplexer(const reg_t &qubits, const std::vector<cmatrix_t> &mmat) {
```

Comment:
Currently there isn't a simple better way to do this unless we completely refactor unitary_state to be a subclass of statevector_state, which has its own issues since it doesn't need a lot of functionality of that class (like measurements). I think this is fine for now

Reply:
I believe that all of the size_t issues have been resolved (most of them were strictly removed through the use of ops.reg[]).  This data cloning issue still exists, as Chris says.